### PR TITLE
Bash completion for spreaper

### PR DESCRIPTION
**Pull request by michaelsembwever**

### CODE REVIEW

**Comment by michaelsembwever** [comment]
@joaquincasares are you any good at knowing how, and testing, that the bash completion script gets installed into the correct location during binary install.

**Comment by joaquincasares** [comment]
@michaelsembwever what gets outputted here is pretty bad, but it works for basic use cases.

https://github.com/joaquincasares/python-bashcomplete

I remember it being slightly buggy in complicated use cases, but the outputted file should be enough to get you up and running. Once you try to customize it you'll quickly realize where the shortfalls are.

Sadly, all I know is what I programmed into this repo and that was after quite a bit of time trying to learn it all. The syntax is what killed me the most! :p 

I just now realized you already had a commit and that looks far better than what my code generates! As far as testing, try: `. ./path/to/completion-file`.

Upon install, I believe this is the ideal place: `/etc/bash_completion`.

Source: https://www.howtoforge.com/how-to-add-bash-completion-in-debian

**Comment by michaelsembwever** [comment]
@joaquincasares 
> I just now realized you already had a commit and that looks far better than what my code generates! …
> Upon install, I believe this is the ideal place: /etc/bash_completion.
> Source: https://www.howtoforge.com/how-to-add-bash-completion-in-debian

Yeah, I wrote that by hand, and it looks up autocompletion by making itself `spreaper` calls. 
So the remaining piece is just ensuring that the script gets put into `/etc/bash_completion.d/`
And when that doesn't work, asking the user to put the line `. /etc/bash_completion.d/spreaper` into their bash profile.

**Comment by joaquincasares** [comment]
Oh, good catch. Yup, `/etc/bash_completion.d/` would be preferred. And that's a good note if they modified their profile for whatever reason.

Your code looks great though! As best as I can tell. :)

**Comment by michaelsembwever** [comment]
@internetstaff have any input on this PR?

**Comment by internetstaff** [comment]
lgtm from an rhel packaging perspective, but I can't test it yet

Thanks for knocking this out - it was on my wishlist / todo!

**Review by adejanovski — CHANGES_REQUESTED** [review]
I've tested the package on Ubuntu after fixing the Makefile and completion works as expected there.

Great addition!

File: src/packaging/Makefile [hunk 1]
```diff
@@ -20,6 +20,7 @@ prepare:
 	cp resource/cassandra-reaper*.yaml build/etc/cassandra-reaper/configs
 	cp ../server/target/cassandra-reaper-$(VERSION).jar build/usr/share/cassandra-reaper/
 	cp bin/* build/usr/local/bin/
+	cp etc/bash_completion.d/spreaper build/etc/bash_completion.d/
```
[the remark this answers is not in the document]
Comment (michaelsembwever) [reply src/packaging/Makefile]: fixed. thanks. ps don't we have any tests for this?

**Review by adejanovski — APPROVED** [review]

**Pull request closed without merging**